### PR TITLE
Demonstration that implementing `getParentExecutable` is necessary

### PR DESCRIPTION
**Pull request by jglick**
Proves that #170 is necessary to retain compatibility with some rare use cases after https://github.com/jenkinsci/jenkins/pull/5880: for example, Pipeline code coverage without https://github.com/jenkinsci/code-coverage-api-plugin/pull/245 or https://github.com/jenkinsci/cobertura-plugin/pull/130.

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java [hunk 1]
```diff
@@ -1219,6 +1234,92 @@ public void getOwnerTaskPermissions() throws Throwable {
...
+    public void accessPermittedOnlyFromCurrentBuild() throws Throwable {
+        sessions.then(r -> {
+            // Adapted from RunningBuildFilePathFilterTest
+            ExtensionList.lookupSingleton(AdminWhitelistRule.class).setMasterKillSwitch(false);
+            WorkflowJob main = r.createProject(WorkflowJob.class, "main");
+            DumbSlave s = r.createOnlineSlave();
+            main.setDefinition(new CpsFlowDefinition("node('" + s.getNodeName() + "') {writeBack()}", true));
+            // Normal case: writing to our own build directory
+            WriteBackStep.controllerFile = new File(main.getBuildDir(), "1/stuff.txt");
+            r.buildAndAssertSuccess(main);
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java:1248]: Fails without `getParentExecutable` override.

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java [hunk 2]
```diff
@@ -1219,6 +1234,92 @@ public void getOwnerTaskPermissions() throws Throwable {
...
+            // Attacks:
+            WriteBackStep.legal = false;
+            // Writing to someone else’s build directory (covered by RunningBuildFilePathFilter)
+            FreeStyleProject other = r.createFreeStyleProject("other");
+            r.buildAndAssertSuccess(other);
+            WriteBackStep.controllerFile = new File(other.getBuildByNumber(1).getRootDir(), "hack");
+            r.buildAndAssertSuccess(main);
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java:1255]: Fails against older Jenkins versions without `RunningBuildFilePathFilter`.

**Comment by jglick** [comment]
This is more `tests` (or `dependencies`) than `enhancement`.

**Comment by car-roll** [comment]
makes sense. I just wanted some kind of indicator that the jenkins version was being bumped, but i can also just add that in the changelog as well

**Pull request merged**